### ISSUE TITLE
Accessibility Issues with Datepicker on Mac and PC

### ISSUE THREAD

**Issue by olavflar** [Altinn/app-frontend-react#2938]
### Description of the bug

On Mac, the datepicker works only in Safari, but when a date is selected and the user presses Enter or tabs to the next element, the focus jumps to the top of the page.
The same happens when tabbing out of the popup window of the navigation.
On Mac, in Firefox and Chrome, it’s not possible to navigate to the popup window using the keyboard.
On PC, it's also not possible to navigate to the popup window in Edge, Chrome, and Firefox.

### Steps To Reproduce

Go to https://ttd.apps.tt02.altinn.no/ttd/frontend-test#/instance/51799569/bfd3e8cf-a489-48ae-bd0d-a4d936d3e04b/Task_2/form
Try to navigate to the datepicker with the keyboard (Tab) in the following environments:
Mac: Safari, Chrome, Firefox
PC: Edge, Chrome, Firefox
Select a date in the popup window.
Press Enter or Tab to move forward.
Expected behavior

### Additional Information

After selecting a date, the focus should logically move to the next element in the focus order.
There should be no unexpected redirection to the top of the page.
Keyboard navigation to the popup window should work across all supported browsers.
